prow/moonraker/client.go

```diff
@@ -95,7 +98,25 @@ func (c *Client) do(method, path string, payload []byte, params map[string]strin
 		q.Set(key, val)
 	}
 	req.URL.RawQuery = q.Encode()
-	return c.httpClient.Do(req)
+	httpClient := c.maybeUpdatedHttpClient()
+	return httpClient.Do(req)
+}
+
+// maybeUpdatedHttpClient returns a new HTTP client with a new one (with
+// a new timeout) if the provided timeout does not match the value already used
+// for the current HTTP client.
+func (c *Client) maybeUpdatedHttpClient() *http.Client {
+	c.Lock()
+	defer c.Unlock()
+	timeout := c.configAgent.Config().Moonraker.ClientTimeout.Duration
+	hc := c.httpClient
```

Comment:
This is taking the pointer to the existing http.Client, which is not what we want. We need a new one. I'd revert to something like you had earlier:

if c.httpClient.Timeout != timeout {
  c.httpClient = &httpClient{Timeout: timeout}
}
return c.httpClient